lib/src/main/scala/spinal/lib/sim/bus/wishbone/WishboneMonitor.scala

```diff
@@ -7,7 +7,8 @@ import spinal.lib.bus.wishbone._
 import scala.collection.mutable._
 
 object WishboneMonitor{
-  def apply(bus : Wishbone, clockdomain: ClockDomain)(callback: (Wishbone) => Unit) = new WishboneMonitor(bus,clockdomain).addCallback(callback)
+  def apply(bus : Wishbone, clockdomain: ClockDomain = null)(callback: (Wishbone) => Unit) =
+    new WishboneMonitor(bus, Option(clockdomain).getOrElse(bus.CYC.clockDomain)).addCallback(callback)
```

Comment:
That is surprising but makes sense in retrospect. I can change those back, but I think this form is still useful for the tests that don't involve any new clocks / clock areas so this logic will hold?

Reply:
This behavior is surprising to many and can easily trip someone up - we've had a few question along those lines. That's why I'd keep the parameter mandatory, if it's automatic it should work. And if you have vastly different clocks it might be easy to see in simulation, if it's just a clock enable it's much harder (I actually found out about this because I wanted to make a similar change to AHB and had to debug for a while a week later)